compiler/semantic/proc.go

```diff
@@ -236,6 +236,9 @@ func semSequential(ctx context.Context, scope *Scope, seq *ast.Sequential, adapt
 func semProc(ctx context.Context, scope *Scope, p ast.Proc, adaptor op.DataAdaptor, head *lakeparse.Commitish) (dag.Op, error) {
 	switch p := p.(type) {
 	case *ast.From:
+		if ctx == nil {
```

Comment:
As mentioned elsewhere, the package context documentation discourages passing a nil context.  Conditioning this on having an adaptor also feels more natural.
```suggestion
		if adaptor == nil {
```